Source/CarthageKit/BuildArguments.swift

```diff
@@ -84,6 +84,11 @@ public struct BuildArguments {
 			args += [ "BITCODE_GENERATION_MODE=\(bitcodeGenerationMode.rawValue)" ]
 		}
 
+		if let carthagePath = NSBundle.mainBundle().executablePath
+			where NSBundle.mainBundle().executableURL?.lastPathComponent?.lowercaseString.rangeOfString("carthage") != nil {
```

Comment:
Fair enough, `lowercaseString.containsString()` it is, assuming we want to allow for the renaming of the binary at all.